### PR TITLE
fix: set HF models to train mode after loading from_pretrained

### PR DESCRIPTION
**Pull request by tonyhoo**
### Issue #, if available:*
#4965 

### Description of changes:*
Fixes issue where HuggingFace backbones remain in eval mode during training due to Lightning v2.2+ preserving submodule training states. This ensures dropout and batch normalization layers function properly during fine-tuning.

### Testing
See [gist](https://gist.github.com/tonyhoo/bba305008740d6c2b0c7970b1d8f6281)
#### Prior to the change
```
AFTER: Model training mode: True
AFTER:   |-- model: training=False  # Bug: HF backbone stuck in eval mode
AFTER:   |-- head: training=True
```

#### Post the change
```
AFTER: Model training mode: True
AFTER:   |-- model: training=True   # Fixed: HF backbone correctly in train mode
AFTER:   |-- head: training=True
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by prateekdesai04** [comment]
/benchmark module=multimodal preset=multimodal_best benchmark=automm-image time_limit=g4_12x

**Comment by tonyhoo** [comment]
/benchmark 482e52fbe0f628f37613477fd173ad7ed5ec9b19 module=multimodal preset=multimodal_best benchmark=automm-image time_limit=g4_12x

**Comment by tonyhoo** [comment]
/benchmark 482e52fbe0f628f37613477fd173ad7ed5ec9b19 module=multimodal preset=multimodal_best benchmark=automm-image time_limit=g4_12x

**Comment by tonyhoo** [comment]
/benchmark 482e52fbe0f628f37613477fd173ad7ed5ec9b19 module=multimodal preset=multimodal_best benchmark=automm-image time_limit=g4_12x

**Comment by tonyhoo** [comment]
/benchmark 482e52fbe0f628f37613477fd173ad7ed5ec9b19 module=multimodal preset=multimodal_best benchmark=automm-image time_limit=g4_12x

**Comment by tonyhoo** [comment]
/benchmark a905c3f96858f068393cad7ec72737a6f12e3a67 module=multimodal preset=multimodal_best benchmark=automm-image time_limit=g4_12x

**Review by suzhoum — APPROVED** [review]
